### ISSUE TITLE
Test failure: baseservices/exceptions/unittests/EHPatternTests/EHPatternTests.cmd

### ISSUE THREAD

**Issue by v-wenyuxu** [dotnet/runtime#98909]
**Failed in:** [runtime-coreclr gcstress-extra 20240224.1](https://dev.azure.com/dnceng-public/public/_build/results?buildId=578457&view=ms.vss-test-web.build-test-results-tab&runId=13881256&resultId=115863&paneView=debug)

**Failed tests:**
```
coreclr osx arm64 Checked gcstress0xc_disabler2r_jitstress2 @ OSX.1200.ARM64.Open
    - baseservices/exceptions/unittests/EHPatternTests/EHPatternTests.cmd
coreclr linux x64 Checked gcstress0xc_disabler2r_jitstress2 @ Ubuntu.2204.Amd64.Open
    - baseservices/exceptions/unittests/EHPatternTests/EHPatternTests.cmd
```

**Error message:**
```
 
Assert failure(PID 37011 [0x00009093], Thread: 1491935 [0x16c3df]): relOffsetOverride != NO_OVERRIDE_OFFSET
    File: /Users/runner/work/1/s/src/coreclr/vm/gcenv.ee.common.cpp:273
    Image: /private/tmp/helix/working/A4C70950/p/corerun

waitpid() returned successfully (wstatus 00000000) WEXITSTATUS 0 WTERMSIG 0
/private/tmp/helix/working/A4C70950/w/9E340918/e/baseservices/exceptions/baseservices-exceptions/../unittests/EHPatternTests/EHPatternTests.sh: line 432: 37011 Abort trap: 6           (core dumped) $LAUNCHER $ExePath "${CLRTestExecutionArguments[@]}"

Return code:      1
Raw output file:      /tmp/helix/working/A4C70950/w/9E340918/uploads/unittests/EHPatternTests/output.txt
Raw output:
BEGIN EXECUTION
/tmp/helix/working/A4C70950/p/corerun -p System.Reflection.Metadata.MetadataUpdater.IsSupported=false -p System.Runtime.Serialization.EnableUnsafeBinaryFormatterSerialization=true EHPatternTests.dll ''
-----------------------------
BaseClassTest
-----------------------------
System.IO.FileNotFoundException: 1
   at BaseClassTest.f2()
   at BaseClassTest.f1()
System.IO.FileNotFoundException: 1
   at BaseClassTest.f1()
   at BaseClassTest.f1()
   at BaseClassTest.Run()
0121: PASS
-----------------------------
TryCatchInFinallyTest
-----------------------------
System.InvalidProgramException: Common Language Runtime detected an invalid program.
   at TryCatchInFinallyTest.Run()
0123456: PASS
-----------------------------
RecurseTest
-----------------------------
System.Exception: 7
   at RecurseTest.DoTest(Int32 level)
System.Exception: 5
   at RecurseTest.DoTest(Int32 level)
System.Exception: 3
   at RecurseTest.DoTest(Int32 level)
System.Exception: 1
   at RecurseTest.DoTest(Int32 level)
876543210: PASS
-----------------------------
ThrowInFinallyNestedInTryTest
-----------------------------
[createdump] Gathering state for process 37011 
[createdump] Crashing thread 16c3df signal 6 (0006)
[createdump] Writing crash report to file /cores/coredump.37011.dmp.crashreport.json
[createdump] Crash report successfully written
[createdump] Writing minidump with heap to file /cores/coredump.37011.dmp
[createdump] Written 359874712 bytes (21965 pages) to core file
[createdump] Target process is alive
[createdump] Dump successfully written in 941ms
Expected: 100
Actual: 134
END EXECUTION - FAILED
Test failed. Trying to see if dump file was created in /cores since 2/24/2024 3:14:55 PM
Processing /cores/coredump.37011.dmp.crashreport.json
Printing stacktrace from '/cores/coredump.37011.dmp.crashreport.json'
Invoking llvm-symbolizer --pretty-print
Errors while running llvm-symbolizer --pretty-print
System.ComponentModel.Win32Exception (2): An error occurred trying to start process 'llvm-symbolizer' with working directory '/private/tmp/helix/working/A4C70950/w/9E340918/e/baseservices/exceptions/baseservices-exceptions'. No such file or directory
   at System.Diagnostics.Process.ForkAndExecProcess(ProcessStartInfo startInfo, String resolvedFilename, String[] argv, String[] envp, String cwd, Boolean setCredentials, UInt32 userId, UInt32 groupId, UInt32[] groups, Int32& stdinFd, Int32& stdoutFd, Int32& stderrFd, Boolean usesTerminal, Boolean throwOnNoExec) in /_/src/libraries/System.Diagnostics.Process/src/System/Diagnostics/Process.Unix.cs:line 499
   at System.Diagnostics.Process.StartCore(ProcessStartInfo startInfo) in /_/src/libraries/System.Diagnostics.Process/src/System/Diagnostics/Process.Unix.cs:line 456
   at CoreclrTestLib.CoreclrTestWrapperLib.TryPrintStackTraceFromCrashReport(String crashReportJsonFile, TextWriter outputWriter)
Finish looking for *.crashreport.json. No new files created.
Test Harness Exitcode is : 1
To run the test:
Set up CORE_ROOT and run.
> /private/tmp/helix/working/A4C70950/w/9E340918/e/baseservices/exceptions/baseservices-exceptions/../unittests/EHPatternTests/EHPatternTests.sh
```

**Stack trace:**
```
   at Xunit.Assert.True(Nullable`1 condition, String userMessage) in /_/src/Microsoft.DotNet.XUnitAssert/src/BooleanAsserts.cs:line 146
   at TestLibrary.OutOfProcessTest.RunOutOfProcessTest(String assemblyPath)
   at Program.<<Main>$>g__TestExecutor140|0_141(StreamWriter tempLogSw, StreamWriter statsCsvSw, <>c__DisplayClass0_0&)
```
Queued | Pipeline | Pipeline_Configuration | OS | Arch | Test | Outcome | Properties | BuildReason
-- | -- | -- | -- | -- | -- | -- | -- | --
2024-02-25T07:39:12.18Z | runtime-coreclr crossgen2-composite gcstress | Checked-gcstress0xc_disabler2r_jitstress2 | osx.1200.arm64.open | arm64 | baseservices/exceptions/unittests/EHPatternTests/EHPatternTests.cmd | Failed | {"BuildId":"578585","DefinitionId":"141","DefinitionName":"runtime-coreclr crossgen2-composite gcstress","configuration":"Checked-gcstress0xc_disabler2r_jitstress2","AzurePipelinesTestRunId":"13885124","System.PhaseAttempt":"1","System.JobName":"__default","operatingSystem":"OSX.1200.ARM64.Open","System.StageAttempt":"1","System.JobId":"4b5575ba-4f05-5198-bc88-a075c735caa5","System.StageName":"Build","Reason":"Schedule","System.JobAttempt":"1","architecture":"arm64","System.PhaseName":"run_test_p1_Composite_osx_arm64_checked","Project":"public","BuildNumber":"20240224.1","CollectionUri":"https://dev.azure.com/dnceng-public/"} | Schedule
2024-02-24T22:40:19.243Z | runtime-coreclr gcstress-extra | Checked-gcstress0xc_disabler2r_jitstress2 | osx.1200.arm64.open | arm64 | baseservices/exceptions/unittests/EHPatternTests/EHPatternTests.cmd | Failed | {"BuildId":"578457","Project":"public","System.StageName":"Build","CollectionUri":"https://dev.azure.com/dnceng-public/","System.JobName":"__default","architecture":"arm64","AzurePipelinesTestRunId":"13881256","System.PhaseAttempt":"1","System.PhaseName":"run_test_p1__osx_arm64_checked","BuildNumber":"20240224.1","operatingSystem":"OSX.1200.ARM64.Open","DefinitionId":"113","System.JobId":"765fd5e9-e47f-5cb0-4cbf-499eaaaee93e","configuration":"Checked-gcstress0xc_disabler2r_jitstress2","System.JobAttempt":"1","Reason":"Schedule","System.StageAttempt":"1","DefinitionName":"runtime-coreclr gcstress-extra"} | Schedule

**Comment by BruceForstall** [comment]
@mangod9

**Comment by janvorli** [comment]
This is one of the issues fixed by #99012, related to the new EH.
